torch/fx/experimental/symbolic_shapes.py

```diff
@@ -21,17 +21,22 @@ def create_contiguous(shape):
         strides.append(dim * strides[-1])
     return list(reversed(strides))
 
-
 def is_symbolic_op(func):
-    return func in [aten.sym_size.default, aten.dim.default, aten.is_contiguous.default, aten.stride.default]
-
+    return func in [aten.sym_size.default, aten.dim.default,
+                    aten.is_contiguous.default, aten.stride.default, aten.sym_numel.default
+                    ]
 
 def handle_symbolic_op(func, args, kwargs):
     assert is_symbolic_op(func)
     if func == torch.ops.aten.sym_size.default:
         return None
     if func == torch.ops.aten.dim.default:
         return len(args[0].shape)
+    if func == torch.ops.aten.sym_numel.default:
+        res = 1
+        for s in args[0].shape:
+            res = res * s
+        return res
```

Comment:
This should turn into a `return None` with an accurate symbolic numel cached on `numel_`. This will prevent us from repeatedly generating the product IR every time numel is queried.

Reply:
I'm not sure about two things before we could do this:

* What would trigger `numel` recomputation on `x.expand()_`?
* If we return `Py_None` it would call `numel_default` which is currently guarded by a) a check that a tensor doesn't have symbolic strides and returns `numel_` which b) we never actually set?

Reply:
We'll need an appropriate hook for cached metadata recomputation when metadata changes.

Also, I think `numel` and `sym_numel` need to be handled asymmetrically. `numel` needs to go into Python and induce a guard when we do the conversion. It's only `sym_numel` that can serve straight from cache.